sansa-ml-spark/src/main/scala/net/sansa_stack/ml/spark/clustering/utils/dataFiltering.scala

```diff
@@ -0,0 +1,65 @@
+package net.sansa_stack.ml.spark.clustering.utils
+
+import org.apache.jena.graph.Triple
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.SparkSession
+import scala.collection.mutable.ArrayBuffer
+
+import net.sansa_stack.ml.spark.clustering.datatypes.appConfig
+import net.sansa_stack.rdf.spark.io.NTripleReader
+
+
+
+class dataFiltering(val spark: SparkSession, val conf: appConfig) extends Serializable {
+
+  val dataRDD: RDD[Triple] = NTripleReader.load(spark, conf.dataset.input).persist()
+
+  /**
+    * Generate triples with related to poi in poiArray, method name not JavaBean format because of side effect with Unit return result
+    * @param poiArray id of pois in Vienna
+    * @param dataRDD RDD containing triples
+    * @param spark SparkSession
+    * @return
+    */
+  def get_triples(poiArray: Array[Long], dataRDD: RDD[Triple], spark: SparkSession) : (RDD[Triple], RDD[Triple]) = {
+    // create an array of subjects related with each poi
+    val subjects = ArrayBuffer[String]()
+    for (i <- 0 until poiArray.length - 1) {
+      subjects ++= createSubjects(poiArray(i))
+    }
+    // RDD[Triple] => RDD[(subject, Triple)]
+    val dataRDDPair = dataRDD.map(f => (f.getSubject.getURI, f)).persist()
+    // create RDD[(subject, subject)] from Array[subjects]
+    val subjectsRDD = spark.sparkContext.parallelize(subjects.toSet.toList).map(f => (f, f)).persist()
+    // get RDD[Triples] with subject in Array[subjects]
+    val viennaTriples = subjectsRDD.join(dataRDDPair).map(f => f._2._2).persist()
+    // find filtered Triples with prediction category, and get their object => RDD[Object]
+    val viennaCatgoriesObjects = viennaTriples.filter(f => f.getPredicate.getURI.equals("http://example.org/def#category")).map(f => f.getObject.getURI).distinct().persist()
```

Comment:
Here we can also use our TripleOps functions (e.g. [contains](https://github.com/SANSA-Stack/SANSA-RDF/blob/develop/sansa-rdf-spark/src/main/scala/net/sansa_stack/rdf/spark/model/package.scala#L132), or [filterPredicates()](https://github.com/SANSA-Stack/SANSA-RDF/blob/develop/sansa-rdf-spark/src/main/scala/net/sansa_stack/rdf/spark/model/package.scala#L88), but we can leave as it is for now and consider in the future.